File: Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDCountry.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOCountry 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  country_id ,country ,last_update  from country";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClCountry country)  
        {
            String stSql = "Select  country_id ,country ,last_update  from country";
            stSql += " Where  country_id='" + country.CountryId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        country.Country = dt.Rows[0]["country"].ToString();
        country.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClCountry country)  
        {
            String stSql = "insert into Country (" 
                       + " country_id ,country ,last_update" 
                           + " )values (" 
            + "'" + country.CountryId + "'" 
                     + ",'" + country.Country + "'" 
                     + ",'" + country.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClCountry country) {
            
            String stSql  =  "delete from Country " 
             + " Where  country_id='" + country.CountryId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClCountry country ) {
            String stSql  =  "update Country SET " 
             + " country = '" + country.Country + "'" 
                             + ",last_update = '" + country.LastUpdate + "'" 
             + " Where  country_id='" + country.CountryId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}

File: Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDCountry.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOCountry
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from country"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( country as ClCountry) as Boolean
             Dim StSql as String = "Select * from country"
            StSql &= " Where  country_id='" & country.dataCountryId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        country.DataCountry = dt.Rows(0)("country").ToString()
        country.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( country as ClCountry)  as Integer
            Dim stSql as String = "insert into Country (" _
                       & " country_id ,country ,last_update" _
                           & " )values (" _
            & "'" & country.dataCountryId & "'" _
                     & ",'" & country.dataCountry & "'" _
                     & ",'" & country.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( country as ClCountry) as Integer
            
            Dim stSql as String =  "delete from Country " _
             & " Where  country_id='" & country.dataCountryId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(country as ClCountry) as Integer
            Dim stSql as String =  "update Country SET " _
             & " country = '" & country.DataCountry & "'" _
                             & ",last_update = '" & country.DataLastUpdate & "'" _
             & " Where  country_id='" & country.dataCountryId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  
